### PR TITLE
#157 Update profile

### LINKED ISSUE #157: Update profile
Allow users (applicants and admins) to update their profile information, ensuring a smooth and secure process for managing personal details.

Key tasks:

1. Allow users to edit personal information, such as name, email, phone number, and profile picture.
2. Ensure input validation for fields like email format, phone number, and required fields.
3. Enable users to upload, update, or remove their profile pictures.
4. Provide options for users to update passwords securely.
5. Include password strength validation and provide password recovery options.
6. Ensure that all changes are saved dynamically and updated in real time.
7. Add a confirmation message or modal once the profile has been successfully updated.

### PR DESCRIPTION
**Pull request by Ismaelmurekezi**
### PR Description
This PR is for allowing user to update their profile information  ensuring a smooth and secure process for managing personal details. 

### Description of the task that was expected to deliver

1. Allow users to edit personal information, such as name, email, phone number, and profile picture.
2. Ensure input validation for fields like email format, phone number, and required fields.
3. Enable users to upload, update, or remove their profile pictures.
4. Provide options for users to update passwords securely.
5. Allow users to edit personal information, such as name, email, phone number, and profile picture.
6. Include password strength validation ,ensure that all changes are saved dynamically and updated in real time.
7. Add a confirmation message or modal once the profile has been successfully updated.

### Locally

- clone this repo and run npm install to install dependencies
- switch to ft-update-profile branch 
- start application by npm run dev
- Create account and login with your credential
- Under profile image on right top corn click it 
- Choose profile
- There you can update any of your information and confirm it by clicking update button

### Deployed 
You follow same steps as those for local 
deployed Link: https://atlp-devpulse-fn-git-ft-upda-10ab6a-devpulses-projects-62c56cf1.vercel.app/

Credentials to use:
Email: admin@example.com
Password: password123  OR

Email: jean@gmail.com
Password: password@123

Issue link: https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/157

###Screenshots
update profile

update profile1

### CODE REVIEW

**Review by Philimuhire — APPROVED** [review]

**Review by mugishaj092 — CHANGES_REQUESTED** [review]
You can improve desigins acccording to figma design

**Review by Ishimwe7 — APPROVED** [review]
Well done @Ismaelmurekezi

**Review by mugishaj092 — APPROVED** [review]

**Review by MnorbertVii — APPROVED** [review]

**Review by SaddockAime — APPROVED** [review]

**Review by mugishaj092 — APPROVED** [review]

**Review by Mugisha146 — APPROVED** [review]

File: src/components/profileDropdown.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ function ProfileDropdown({
                 <div className="flex flex-col w-full gap-[5px] cursor-pointer">
                   <Link
                     onClick={handleShowProfileDropdown}
-                    to="/dashboard/profile"
+                    to="update-profile"
```
Comment (ncutixavier) [inline src/components/profileDropdown.tsx]: Why did you change redirection link?

File: src/components/updateUserProfile/imagePreview.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { toast } from "react-toastify";
+import { TuserSchema } from "../../utils/userSchema";
+
+interface ImagePreviewProps {
+  previewImage: string | null;
+  form: UseFormReturn<TuserSchema>;
+}
+
+const DEFAULT_IMAGE =
+  "https://t4.ftcdn.net/jpg/01/24/65/69/360_F_124656969_x3y8YVzvrqFZyv3YLWNo6PJaC88SYxqM.jpg";
```
Comment (ncutixavier) [inline src/components/updateUserProfile/imagePreview.tsx]: Let use our local images only

File: src/components/updateUserProfile/useImagePreview.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { UseFormReturn } from "react-hook-form";
+import {
+  handleFilePreview,
+  handleImageUpload,
+} from "../../utils/imageuploadUtil";
+import { TuserSchema } from "../../utils/userSchema";
+import { toast } from "react-toastify";
+
+const DEFAULT_IMAGE =
+  "https://t4.ftcdn.net/jpg/01/24/65/69/360_F_124656969_x3y8YVzvrqFZyv3YLWNo6PJaC88SYxqM.jpg";
```
Comment (ncutixavier) [inline src/components/updateUserProfile/useImagePreview.tsx]: -Let use our local images only
- Avoid outside images links

File: src/utils/Profileutils.ts
Comment (ncutixavier) [inline src/utils/Profileutils.ts:1]: File naming is wrong

**Review by ncutixavier — COMMENTED** [review]
- Attention to file naming, follow camelCase structure
- Don't use outside images links

File: src/components/profileDropdown.tsx (hunk 1 above)
Comment (Ismaelmurekezi) [reply src/components/profileDropdown.tsx]: Thanks, I solved that path to its original  path, I was using this path to reach update profile page,because there wasn't profile page to reach update-profile page , but thanks to you I have solved that.

File: src/components/updateUserProfile/imagePreview.tsx (hunk 2 above)
Comment (Ismaelmurekezi) [reply src/components/updateUserProfile/imagePreview.tsx]: Yeah , now I'm using local images

File: src/utils/Profileutils.ts
Comment (Ismaelmurekezi) [reply src/utils/Profileutils.ts:1]: I didn't see that,but I have fixed that

File: src/components/updateUserProfile/useImagePreview.tsx (hunk 3 above)
Comment (Ismaelmurekezi) [reply src/components/updateUserProfile/useImagePreview.tsx]: I have changed that
